### PR TITLE
KOGITO-7798 Refactor setup branch job

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-7798

Allow to update the version also on main branch
- Renamed init-branch -> setup-branch
- Prepare release job will call the setup branch on main branch as well with correct version

Related PRs:
- https://github.com/kiegroup/kogito-pipelines/pull/602
- https://github.com/kiegroup/drools/pull/4713
- https://github.com/kiegroup/kogito-runtimes/pull/2502
- https://github.com/kiegroup/kogito-apps/pull/1471
- https://github.com/kiegroup/kogito-examples/pull/1414
- https://github.com/kiegroup/kogito-images/pull/1340
- https://github.com/kiegroup/kogito-operator/pull/1288
- https://github.com/kiegroup/kogito-docs/pull/214

### CODE REVIEW

File: dsl/seed/jenkinsfiles/Jenkinsfile.release.prepare [hunk 1]
```diff
@@ -27,8 +27,7 @@ pipeline {
                 script {
                     checkout scm
 
-                    List projects = getEcosystemProjects().collect { it.name }
-                    String projectListStr = projects.collect { "${getRepoNameCamelCase(it)} ${getVersionFromRepository(it)}" }.join(' / ')
+                    String projectListStr = getEcosystemProjects().collect { it.name }.collect { "${getRepoNameCamelCase(it)} ${getVersionFromRepository(it)}" }.join(' / ')
```
Comment (Ginxo) [inline dsl/seed/jenkinsfiles/Jenkinsfile.release.prepare]:
```suggestion
                    String projectListStr = getEcosystemProjects().collect { "${getRepoNameCamelCase(it.name)} ${getVersionFromRepository(it.name)}" }.join(' / ')
```
right? :thinking:

**Review by Ginxo — APPROVED** [review]
LGTM, just one minor technical suggestion

File: dsl/seed/jenkinsfiles/Jenkinsfile.release.prepare (hunk 1 above)
Comment (radtriste) [reply dsl/seed/jenkinsfiles/Jenkinsfile.release.prepare]: looks good indeed :)

**Comment by radtriste** [comment]
OP checks resolution will be done in https://github.com/kiegroup/kogito-pipelines/pull/646

**Pull request merged**